lighthouse-core/audits/seo/crawlable-anchors.js

```diff
@@ -0,0 +1,108 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit.js');
+const i18n = require('../../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on whether anchors have hyperlinks which can be crawled by search engines. This descriptive title is shown when all hyperlinks on the page are crawlable. */
+  title: 'Anchors have crawlable hyperlinks',
+  /** Descriptive title of a Lighthouse audit that provides detail on whether anchors have hyperlinks which can be crawled by search engines. This descriptive title is shown when there are hyperlinks which are not crawlable by search engines. */
+  failureTitle: 'Anchors do not have crawlable hyperlinks',
+  /** Description of a Lighthouse audit that tells the user why hyperlinks should be crawlable. This is displayed after a user expands the section to see more. 'Learn More' becomes link text to additional documentation. */
+  description: 'Search engines use hyperlinks to crawl websites. Ensure that the `href` attribute of anchor elements links to an appropriate destination, so more pages of the site can be discovered. [Learn More](https://support.google.com/webmasters/answer/9112205)',
+  /** Label for a column in a data table; entries will be the HTML anchor elements that failed the audit. Anchors are DOM elements that are links. */
+  columnFailingAnchors: 'Uncrawlable Anchor Element',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class CrawlableAnchors extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'crawlable-anchors',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['AnchorElements'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit({AnchorElements: anchorElements}) {
+    const failingAnchors = anchorElements.filter(({
+      rawHref,
+      hasClickHandler,
+      onclick = '',
+      name = '',
+    }) => {
+      onclick = onclick.replace( /\s/g, '');
+      name = name.trim();
+      rawHref = rawHref.replace( /\s/g, '');
+
+      const windowLocationRegExp = /window.location=/;
+      const windowOpenRegExp = /window.open\(/;
+      const javaScriptVoidRegExp = /javascript:void(\(|)0(\)|)/;
+
+      if (rawHref.startsWith('file:')) {
+        return true;
+      }
+
+      if (windowLocationRegExp.test(onclick)) {
```

Comment:
Happy to turn all these into one liners which might not make this a long function body

```js
if (windowLocationRegExp.test(onclick)) return true;
```

Reply:
If they can all work as one-liners I do have a *personal* preference for it to condense things a bit, but we're not totally consistent on that as a team so up to you